src/btree/bt_curnext.c

```diff
@@ -888,7 +888,12 @@ __wt_btcur_next_prefix(WT_CURSOR_BTREE *cbt, WT_ITEM *prefix, bool truncating)
  *     Move to the next record in the tree.
  */
 int
-__wt_btcur_next(WT_CURSOR_BTREE *cbt, bool truncating)
+__wt_btcur_next(WT_CURSOR_BTREE *cbt, bool truncating, bool evict_reposition)
```

Comment:
The way evict_reposition is used makes me think the layering isn't quite right. Having all the callers hard code a true/false feels a bit wrong.

Is it actually a cursor state? i.e: sometimes we do want to do the repositioning dance when a hot page is encountered and sometimes we don't? If so do you think it'd make sense to add a flag to the WT_CURSOR_BTREE instead of passing the parameter around?

Reply:
Since `__wt_btcur_search`,   `__wt_btcur_next` and  `__wt_btcur_prev` is called from multiple places. The purpose of the boolean is to only reposition if these functions are called from the file cursor(where the boolean is set true).

For instance, `__wt_btcur_next` is called as part of truncate and I am thinking that we wouldn't want to call the reposition function in that case?

Reply:
Updated in [f62c0c1](https://github.com/wiredtiger/wiredtiger/pull/7687/commits/f62c0c12a1c89ff39d5faa9fc636e996db449c64)

Reply:
I still don't see why this is tied to file based cursors. In the end, pretty much all cursor operations end up using file based cursors (they are like the base level building block for cursors).

I'm concerned that the way this is layered in will mean that the functionality isn't triggered when expected.

It feels like it should be layered into the `CURSOR_API_CALL` and `API_END` macros - but I took a look at those, and they are already too complex, so layering this into there as well doesn't make sense.
I've got a concrete suggestion about a different approach - but it would be best to talk it through in person.